src/main/scala/ai/privado/audit/DataElementDiscovery.scala

```diff
@@ -41,9 +74,271 @@ object DataElementDiscoveryUtils {
+      case Success(cpg) => {
+        println(cpg.identifier.filter(i => i.name.length > 2).l.size)
+        cpg.identifier
+          .filter(i => i.name.length > 2)
+          .filter(i => !filterCommonVarsStartsWithArr.exists(xx => i.name.startsWith(xx)))
+          .filter(i => !i.name.matches(filterCommonLangTypes))
+          .filter(i => !i.name.matches(filterCommonVars))
+          .filter(i => !i.name.matches(DataElementDiscoveryUtils.getLanguageSpecificFilters(lang)))
+          .dedup
+          .l
+      }
+      case Failure(ex) => {
+        ex.printStackTrace()
+        List[Identifier]()
+      }
+    }
+
+    val addedIdentifiers = mutable.Set[String]()
+    identifiers.foreach(identifier => {
+      val nodeType = AuditReportConstants.ELEMENT_DISCOVERY_NODE_TYPE_IDENTIFIER
+      val (path, lineNumber, nodeUniqueId) =
+        DataElementDiscoveryUtils.getNodeLocationAndUniqueId(identifier, identifier.name, nodeType)
+
+      val identifierUniqueKey = s"${identifier.typeFullName}$path${identifier.name}"
+      val sourceRuleId =
+        identifier.tag.nameExact(Constants.id).value.headOption.getOrElse(AuditReportConstants.AUDIT_EMPTY_CELL_VALUE)
+
+      if (
+        identifier.name.nonEmpty && !identifier.name
+          .matches(AuditReportConstants.JS_ELEMENT_DISCOVERY_TYPE_EXCLUDE_REGEX)
+        && !identifier.name
+          .matches(AuditReportConstants.JS_ELEMENT_DISCOVERY_EXCLUDE_PARAMS_REGEX)
+        && !identifier.name.matches(AuditReportConstants.JS_ELEMENTS_TO_BE_EXCLUDED)
+        && !addedIdentifiers.contains(identifierUniqueKey)
+      )
+        addedIdentifiers.add(identifierUniqueKey)
+      workbookResult += List(
+        identifier.typeFullName,
+        identifier.file.name.headOption.getOrElse(AuditReportConstants.AUDIT_EMPTY_CELL_VALUE),
+        "0.0",
+        identifier.name,
+        identifier.typeFullName,
+        sourceRuleId.startsWith("Data.Sensitive.").toString,
+        sourceRuleId,
+        AuditReportConstants.AUDIT_EMPTY_CELL_VALUE,
+        AuditReportConstants.AUDIT_EMPTY_CELL_VALUE,
+        AuditReportConstants.AUDIT_EMPTY_CELL_VALUE,
+        lineNumber,
+        nodeUniqueId,
+        nodeType
+      )
+    })
+
+    workbookResult
+  }
+
+  // Add FieldIdentifiers in workbookResult
+  def getFieldAccessIdentifier(
+    xtocpg: Try[Cpg],
+    workbookResult: ListBuffer[List[String]],
+    lang: Language
+  ): ListBuffer[List[String]] = {
+    val fieldIdentifiers = xtocpg match {
+      case Success(cpg) => {
+        cpg.fieldAccess.fieldIdentifier
+          .filter(i => i.canonicalName.length > 2)
+          .filter(i => !filterCommonVarsStartsWithArr.exists(xx => i.canonicalName.startsWith(xx)))
+          .filter(i => !i.canonicalName.matches(filterCommonLangTypes))
+          .filter(i => !i.canonicalName.matches(filterCommonVars))
+          .filter(i => !i.canonicalName.matches(DataElementDiscoveryUtils.getLanguageSpecificFilters(lang)))
```

Comment:
Can we add some comments in the code explaining these filters? Looks like there are many, else we could add methods with descriptive names.